src/test/rustdoc-ui/intra-doc/unknown-disambiguator.stderr

```diff
@@ -0,0 +1,40 @@
+error: unknown disambiguator `foo`
+  --> $DIR/unknown-disambiguator.rs:1:17
+   |
+LL | //! Linking to [foo@banana] and [`bar@banana!()`].
+   |                 ^^^^^^^^^^
```

Comment:
Is there a way to get this to only point to `foo`? Then you wouldn't need the extra `note`.
```suggestion
LL | //! Linking to [foo@banana] and [`bar@banana!()`].
   |                 ^^^
```
I think maybe you could return the span as part of the error in `from_str`? Not sure how you'd tell if there were backticks stripped though.

Reply:
> I think maybe you could return the span as part of the error in `from_str`?

Yeah, that seems like a good way to approach it.

> Not sure how you'd tell if there were backticks stripped though.

I think we'd return the range relative to what's passed in to `from_str` and then the caller would adjust it.

Reply:
It works! 🎉 

```
error: unknown disambiguator `bar`
  --> $DIR/unknown-disambiguator.rs:1:35
   |
LL | //! Linking to [foo@banana] and [`bar@banana!()`].
   |                                   ^^^
```